src/lib/Type.hs

```diff
@@ -276,6 +276,7 @@ exprEffs expr = case expr of
       MAsk    -> S.singleton (Reader, h)
       MTell _ -> S.singleton (Writer, h)
       where RefTy (Var (h:>_)) _ = getType ref
+    FFICall _ _ _ -> S.singleton (State, theWorld)
```

Comment:
This is actually very helpful. Previously it was difficult to be sure that we don't end up accidentally putting FFICalls in GPU kernels, but now the criterion is simple -- no IO effect nor `runIO` is allowed there.